lib/main.dart

```diff
@@ -11,21 +12,63 @@ import 'package:pispapp/ui/theme/light_theme.dart';
 import 'controllers/app/account_controller.dart';
 import 'controllers/app/auth_controller.dart';
 import 'controllers/ephemeral/dashboard_controller.dart';
+import 'controllers/ephemeral/local_auth_controller.dart';
 import 'controllers/ephemeral/login_controller.dart';
 import 'controllers/ephemeral/profile_controller.dart';
 import 'controllers/ephemeral/splash_controller.dart';
 
 void main() {
+  // Ensures flutter binding is created even before runApp() so
+  // binary messenger can be used for async code
+  WidgetsFlutterBinding.ensureInitialized();
+
   initControllers();
 
-  runApp(GetMaterialApp(
-    debugShowCheckedModeBanner: false,
-    initialRoute: '/',
-    theme: appThemeData,
-    defaultTransition: Transition.fade,
-    getPages: AppPages.pages,
-    home: SplashScreen(),
-  ));
+  runApp(LifecycleAwareApp());
+}
+
+// Wrapper class to allow app-wide lifecycle listening
+class LifecycleAwareApp extends StatefulWidget {
+  @override
+  _LifecycleAwareAppState createState() => _LifecycleAwareAppState();
+}
+
+class _LifecycleAwareAppState extends State<LifecycleAwareApp> with WidgetsBindingObserver {
+  @override
+  void initState() {
+    WidgetsBinding.instance.addObserver(this);
+    // Show verification screen when app first starts
+    Get.find<LocalAuthController>().appWasResumed();
+    super.initState();
+  }
+
+  @override
+  void dispose() {
+    WidgetsBinding.instance.removeObserver(this);
+    super.dispose();
+  }
+
+  @override
+  void didChangeAppLifecycleState(AppLifecycleState state) {
+    if (state == AppLifecycleState.resumed) {
```

Comment:
I think it would be better to use switch-case rather than if-else for handling enum